src/dense_graph.rs

```diff
@@ -0,0 +1,741 @@
+use std::collections::hash_set;
+use std::marker::PhantomData;
+use std::ops::{Index, IndexMut};
+use std::cmp;
+
+use dense_mapping::{DenseMapping, IdIterator};
+use fixedbitset::FixedBitSet;
+
+use {
+    Directed,
+    Direction,
+    EdgeType,
+    Incoming,
+    IntoWeightedEdge,
+    //Outgoing,
+    Undirected,
+};
+
+pub use graph::{
+    DefaultIx,
+    EdgeIndex,
+    IndexType,
+    NodeIndex,
+};
+
+use visit::{
+    GetAdjacencyMatrix,
+    GraphBase,
+    IntoNeighbors,
+    IntoNeighborsDirected,
+    IntoNodeIdentifiers,
+    NodeCount,
+    Visitable
+};
+
+pub use graph::{node_index, edge_index};
+
+type OptionVec<T> = Vec<Option<T>>;
+
+const INITIAL_RESIZE: usize = 10;
+
+fn ensure_len<T>(vec: &mut OptionVec<T>, min_len: usize) where T: Clone {
+    if vec.len() <= min_len {
+        let next_power_of_2_len = (min_len + 2) & !1;
```

Comment:
I don't understand how this works.

I would probably elect to let the vec decide pace to grow, and just ask it to grow to the min_len, and fill it with None until its resulting capacity.